.test/run.sh

```diff
@@ -625,52 +640,60 @@ fi
-	do
-		sleep 1
-		c=$(( c - 1 ))
-	done
-	mariadbclient -u $mariadb_replication_user -p$mariadb_replication_password replcheck --batch --skip-column-names -e 'show create table t1'
-
-	docker exec --user mysql -i \
-		"$cname" \
-		$mariadb \
-		-e "SHOW SLAVE STATUS\G"
-	killoff
+	# Chesck $mariadb_replication_user get created or not
+	if checkUserExistInMariaDB $mariadb_replication_user ;
+	then
+		grants=$(mariadbclient -u $mariadb_replication_user -p$mariadb_replication_password -e "SHOW GRANTS")
+		[[ "${grants/SLAVE/REPLICA}" =~ "GRANT REPLICATION REPLICA ON *.* TO \`$mariadb_replication_user\`@\`%\`" ]] || die "I wasn't created how I was expected: got $grants"
+
+		mariadbclient -u root --batch --skip-column-names -e 'create table t(i int)' replcheck
+		readarray -t vals < <(mariadbclient -u root --batch --skip-column-names -e 'show master status\G' replcheck)
+		lastfile="${vals[1]}"
+		pos="${vals[2]}"
+		[[ "$lastfile" = my-mariadb-bin.00000[12] ]] || die "too many binlog files"
+		[ "$pos" -lt 500 ] || die 'binary log too big'
+		docker exec "$cid" ls -la /var/lib/mysql/my-mariadb-bin.000001
+		docker exec "$cid" sh -c '[ $(wc -c < /var/lib/mysql/my-mariadb-bin.000001 ) -gt 2500 ]' && die 'binary log 1 too big'
+		docker exec "$cid" sh -c "[ \$(wc -c < /var/lib/mysql/$lastfile ) -gt $pos ]" && die 'binary log 2 too big'
+
+		master_host=$cname
+		master_cid=$cid
+
+		cname="mariadb-container-$RANDOM-$RANDOM"
+		runandwait \
+			-e MASTER_HOST="$master_host" \
+			-e MARIADB_RANDOM_ROOT_PASSWORD=1 \
+			-e MARIADB_REPLICATION_MODE="replica" \
+			-e MARIADB_REPLICATION_USER="$mariadb_replication_user" \
+			-e MARIADB_REPLICATION_PASSWORD="$mariadb_replication_password" \
+			-e MARIADB_MYSQL_LOCALHOST_USER=1 \
+			-e MARIADB_MYSQL_LOCALHOST_GRANTS="REPLICA MONITOR" \
+			--network=container:"$master_host" \
+			--health-cmd='healthcheck.sh --su-mysql --replication_io --replication_sql --replication_seconds_behind_master=0 --replication' \
+			--health-interval=3s \
+			"$image" --server-id=3001 --port 3307
+
+		c="${DOCKER_LIBRARY_START_TIMEOUT:-10}"
+		until docker exec "$cid" healthcheck.sh --su-mysql --connect --replication_io --replication_sql --replication_seconds_behind_master=0 --replication || [ "$c" -eq 0 ]
+		do
+			sleep 1
+			c=$(( c - 1 ))
+		done
+		mariadbclient -u $mariadb_replication_user -p$mariadb_replication_password replcheck --batch --skip-column-names -e 'show create table t1'
+
+		docker exec --user mysql -i \
+			"$cname" \
+			$mariadb \
+			-e "SHOW SLAVE STATUS\G"
+		killoff
 
-	cid=$master_cid
-	killoff
+		cid=$master_cid
+		killoff
+	else
+		# Removes created container for MARIADB_REPLICATION_MODE="master"
+		killoff
+		echo -e "Test: It seems like docker image version is not required to check this replication test, Only the versions greater than 11.1 will require this replication test.\n"
```

Comment:
Why? The same code will work with 10.5+, and with a small bit of additional effort 10.3 and 10.4.

Reply:
I thought this enhancement is gonna be in upcoming versions of docker images that's why I wrote that message. BTW I've changed message now.